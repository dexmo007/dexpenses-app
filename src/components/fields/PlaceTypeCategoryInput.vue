<template>
  <v-autocomplete
    v-bind="$attrs"
    :value="value"
    @input="$emit('input', $event)"
    :items="placeTypeCategories"
    item-text="name"
    item-value="value"
    menu-props="auto"
    label="Place Type Category"
    hide-details
    single-line
  ></v-autocomplete>
</template>
<script>
import { placeTypeMappings } from '@dexpenses/core';
import { prettifySnakeCase } from '@/util/string';
import { alphabeticallyBy } from '@/util/sort';

export default {
  props: ['value'],
  data() {
    return {
      placeTypeCategories: [...new Set(Object.values(placeTypeMappings))]
        .filter(c => !!c)
        .map(placeTypeCategory => ({
          value: placeTypeCategory,
          name: prettifySnakeCase(placeTypeCategory),
        }))
        .sort(alphabeticallyBy('name')),
    };
  },
};
</script>
