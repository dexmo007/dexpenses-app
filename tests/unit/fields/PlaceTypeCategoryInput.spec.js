import { mount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';
import PlaceTypeCategoryInput from '@/components/fields/PlaceTypeCategoryInput.vue';

Vue.use(Vuetify);

describe('PlaceTypeCategoryInput.vue', () => {
  it('should renders correctly', () => {
    const wrapper = mount(PlaceTypeCategoryInput);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correctly with value', () => {
    const wrapper = mount(PlaceTypeCategoryInput, {
      propsData: {
        value: 'food',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should trigger input event if user selects new item', () => {
    const wrapper = mount(PlaceTypeCategoryInput);
    wrapper.find('.v-select__slot').trigger('click');
    wrapper.find('.v-list__tile').trigger('click'); // click first item

    expect(wrapper.emitted().input).toBeTruthy();
    expect(wrapper.emitted().input[0]).toHaveLength(1);
    expect(wrapper.emitted().input[0][0]).toEqual('barber');
  });

  it('should not trigger input event if new value is set programmatically', () => {
    const wrapper = mount(PlaceTypeCategoryInput, {
      propsData: {
        value: 'food',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    wrapper.setProps({
      value: 'barber',
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.emitted().input).toBeFalsy();
  });

  it('should update v-model correctly', () => {
    const wrapper = mount({
      template: '<PlaceTypeCategoryInput v-model="modelValue"/>',
      data: () => ({
        modelValue: null,
      }),
      components: {
        PlaceTypeCategoryInput,
      },
    });

    wrapper.find('.v-select__slot').trigger('click');
    wrapper.find('.v-list__tile').trigger('click'); // click first item

    expect(wrapper.vm.modelValue).toEqual('barber');
    expect(wrapper.element).toMatchSnapshot();
  });
});
