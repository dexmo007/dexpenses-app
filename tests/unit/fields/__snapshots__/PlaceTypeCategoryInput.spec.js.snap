// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceTypeCategoryInput.vue should not trigger input event if new value is set programmatically 1`] = `
<div
  class="v-input v-text-field v-text-field--single-line v-select v-autocomplete v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
>
  <div
    class="v-input__control"
  >
    <div
      class="v-input__slot"
    >
      <div
        class="v-select__slot"
      >
        <input
          aria-label="Place Type Category"
          autocomplete="off"
          role="combobox"
          type="text"
        />
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              arrow_drop_down
            </i>
          </div>
        </div>
      </div>
      <div
        class="v-menu"
      >
        <div
          class="v-menu__content theme--light v-menu__content--auto v-autocomplete__content"
          style="max-height: 200px; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
        >
          <div
            class="v-select-list v-card theme--light"
          >
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Barber
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Car
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Clothing
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Education
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Entertainment
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="primary--text"
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Food
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Health
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Household
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Pet
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Relocation
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Tech
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Transport
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Travel
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlaceTypeCategoryInput.vue should not trigger input event if new value is set programmatically 2`] = `
<div
  class="v-input v-text-field v-text-field--single-line v-select v-autocomplete v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
>
  <div
    class="v-input__control"
  >
    <div
      class="v-input__slot"
    >
      <div
        class="v-select__slot"
      >
        <input
          aria-label="Place Type Category"
          autocomplete="off"
          role="combobox"
          type="text"
        />
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              arrow_drop_down
            </i>
          </div>
        </div>
      </div>
      <div
        class="v-menu"
      >
        <div
          class="v-menu__content theme--light v-menu__content--auto v-autocomplete__content"
          style="max-height: 200px; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
        >
          <div
            class="v-select-list v-card theme--light"
          >
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                class="primary--text"
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Barber
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Car
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Clothing
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Education
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Entertainment
                    </div>
                  </div>
                </a>
              </div>
              <div
                class=""
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Food
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Health
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Household
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Pet
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Relocation
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Tech
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Transport
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Travel
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlaceTypeCategoryInput.vue should render correctly with value 1`] = `
<div
  class="v-input v-text-field v-text-field--single-line v-select v-autocomplete v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
>
  <div
    class="v-input__control"
  >
    <div
      class="v-input__slot"
    >
      <div
        class="v-select__slot"
      >
        <input
          aria-label="Place Type Category"
          autocomplete="off"
          role="combobox"
          type="text"
        />
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              arrow_drop_down
            </i>
          </div>
        </div>
      </div>
      <div
        class="v-menu"
      >
        <div
          class="v-menu__content theme--light v-menu__content--auto v-autocomplete__content"
          style="max-height: 200px; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
        >
          <div
            class="v-select-list v-card theme--light"
          >
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Barber
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Car
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Clothing
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Education
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Entertainment
                    </div>
                  </div>
                </a>
              </div>
              <div
                class="primary--text"
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Food
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Health
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Household
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Pet
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Relocation
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Tech
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Transport
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Travel
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlaceTypeCategoryInput.vue should renders correctly 1`] = `
<div
  class="v-input v-text-field v-text-field--single-line v-select v-autocomplete v-input--hide-details theme--light"
>
  <div
    class="v-input__control"
  >
    <div
      class="v-input__slot"
    >
      <div
        class="v-select__slot"
      >
        <label
          aria-hidden="true"
          class="v-label theme--light"
          style="left: 0px; position: absolute;"
        >
          Place Type Category
        </label>
        <input
          aria-label="Place Type Category"
          autocomplete="off"
          role="combobox"
          type="text"
        />
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              arrow_drop_down
            </i>
          </div>
        </div>
      </div>
      <div
        class="v-menu"
      >
        <div
          class="v-menu__content theme--light v-menu__content--auto v-autocomplete__content"
          style="max-height: 200px; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
        >
          <div
            class="v-select-list v-card theme--light"
          >
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Barber
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Car
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Clothing
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Education
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Entertainment
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Food
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Health
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Household
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Pet
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Relocation
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Tech
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Transport
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Travel
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlaceTypeCategoryInput.vue should update v-model correctly 1`] = `
<div
  class="v-input v-text-field v-text-field--single-line v-select v-autocomplete v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--is-focused theme--light primary--text"
>
  <div
    class="v-input__control"
  >
    <div
      class="v-input__slot"
    >
      <div
        class="v-select__slot"
      >
        <input
          aria-label="Place Type Category"
          autocomplete="off"
          role="combobox"
          type="text"
        />
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light primary--text"
            >
              arrow_drop_down
            </i>
          </div>
        </div>
      </div>
      <div
        class="v-menu"
      >
        <div
          class="v-menu__content theme--light v-menu__content--auto v-autocomplete__content"
          style="max-height: 200px; min-width: 0px; max-width: auto; top: 12px; left: 12px; transform-origin: top left; z-index: 8; display: none; z-index: 8;"
        >
          <div
            class="v-select-list v-card theme--light"
          >
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                class="primary--text"
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Barber
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Car
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Clothing
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Education
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Entertainment
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Food
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Health
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Household
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Pet
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Relocation
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Tech
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Transport
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      Travel
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
